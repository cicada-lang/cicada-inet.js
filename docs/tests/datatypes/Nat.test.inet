require "Nat.inet"

import Nat, zero, zero as z, add1, add from "Nat.inet"