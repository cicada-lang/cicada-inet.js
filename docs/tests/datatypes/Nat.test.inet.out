(zero₃)-value!target-(add₅)
(zero₂)-value addend-(add₅)
(add₅)-return

(zero₂)-value

(add1₄)-value!target-(add₆)
(add1₃)-value addend-(add₆)
(zero₅)-value prev-(add1₄)
(zero₄)-value prev-(add1₃)
(add₆)-return

(add1₃)-value prev-(add1₅)
(zero₄)-value prev-(add1₃)
(add1₅)-value

(add1₉)-value!target-(add₉)
(add1₇)-value!target-(add₈)
(add₈)-return addend-(add₁₀)
(add₉)-return target-(add₁₀)
(add1₈)-value addend-(add₉)
(zero₉)-value prev-(add1₉)
(zero₈)-value prev-(add1₈)
(add1₆)-value addend-(add₈)
(zero₇)-value prev-(add1₇)
(zero₆)-value prev-(add1₆)
(add₁₀)-return

(add1₆)-value prev-(add1₁₃)
(add1₁₃)-value prev-(add1₁₂)
(add1₁₂)-value prev-(add1₁₁)
(zero₆)-value prev-(add1₆)
(add1₁₁)-value

(add1₁₄)-value!target-(add₁₅)
(zero₁₀)-value addend-(add₁₅)
(zero₁₁)-value prev-(add1₁₄)
(add₁₅)-return

(zero₁₀)-value prev-(add1₁₅)
(add1₁₅)-value

(add1₁₆)-value prev-(add1₂₁)
(add1₂₁)-value prev-(add1₂₀)
(zero₁₂)-value prev-(add1₁₆)
(add1₂₀)-value

(add1₂₅)-value!target-(add₂₂)
(add₂₂)-return target-(nat_erase₅)
(add1₂₄)-value addend-(add₂₂)
(zero₁₈)-value prev-(add1₂₅)
(zero₁₇)-value prev-(add1₂₄)
(zero₁₉)-value

(zero₁₉)-value

(add1₃₂)-value!target-(add₂₄)
(add₂₄)-return target-(nat_dup₅)
(add1₃₁)-value addend-(add₂₄)
(zero₂₅)-value prev-(add1₃₂)
(zero₂₄)-value prev-(add1₃₁)
(nat_dup₅)-second
(nat_dup₅)-first

(zero₂₇)-value prev-(add1₃₇)
(zero₂₆)-value prev-(add1₃₆)
(add1₃₇)-value prev-(add1₃₅)
(add1₃₆)-value prev-(add1₃₄)
(add1₃₄)-value
(add1₃₅)-value

(add1₄₃)-value!target-(add₂₈)
(add1₄₁)-value!target-(add₂₇)
(add₂₇)-return mulend-(mul₅)
(add₂₈)-return target-(mul₅)
(add1₄₂)-value addend-(add₂₈)
(zero₃₄)-value prev-(add1₄₃)
(zero₃₃)-value prev-(add1₄₂)
(add1₄₀)-value addend-(add₂₇)
(zero₃₂)-value prev-(add1₄₁)
(zero₃₁)-value prev-(add1₄₀)
(mul₅)-return

(zero₃₉)-value prev-(add1₅₇)
(add1₅₁)-value prev-(add1₅₄)
(add1₅₇)-value prev-(add1₅₁)
(add1₅₄)-value prev-(add1₄₈)
(add1₄₈)-value

(add1₆₃)-value!target-(add₄₂)
(add1₆₂)-value!target-(add₄₁)
(add1₆₀)-value!target-(add₄₀)
(add1₅₉)-value!target-(add₃₉)
(add₄₀)-return mulend-(mul₁₀)
(add₄₂)-return target-(mul₁₀)
(add₄₁)-return addend-(add₄₂)
(zero₄₉)-value prev-(add1₆₃)
(add1₆₁)-value addend-(add₄₁)
(zero₄₈)-value prev-(add1₆₂)
(zero₄₇)-value prev-(add1₆₁)
(add₃₉)-return addend-(add₄₀)
(zero₄₆)-value prev-(add1₆₀)
(add1₅₈)-value addend-(add₃₉)
(zero₄₅)-value prev-(add1₅₉)
(zero₄₄)-value prev-(add1₅₈)
(mul₁₀)-return

(zero₅₆)-value prev-(add1₉₄)
(add1₈₅)-value prev-(add1₉₁)
(add1₇₂)-value prev-(add1₈₈)
(add1₉₄)-value prev-(add1₈₅)
(add1₉₁)-value prev-(add1₈₂)
(add1₈₈)-value prev-(add1₇₉)
(add1₈₂)-value prev-(add1₇₅)
(add1₇₉)-value prev-(add1₆₉)
(add1₇₅)-value prev-(add1₇₂)
(add1₆₉)-value

(add1₁₀₁)-value prev-(add1₁₀₄)
(zero₆₈)-value prev-(add1₁₀₁)
(add1₁₀₄)-value

(add1₁₁₀)-value prev-(add1₁₀₉)
(zero₇₁)-value prev-(add1₁₁₀)
(add1₁₀₉)-value

(add1₁₁₁)-value prev-(add1₁₂₀)
(add1₁₂₀)-value prev-(add1₁₁₈)
(zero₇₃)-value prev-(add1₁₁₁)
(add1₁₁₈)-value

